lib/metadata.js

```diff
@@ -174,16 +174,17 @@ Metadata.prototype.updateOidcMetadata = function updateOidcMetadata(doc, next) {
 Metadata.prototype.generateOidcPEM = function generateOidcPEM(kid) {
   const keys = this && this.oidc && Array.isArray(this.oidc.keys) ? this.oidc.keys : null;
   let pubKey = null;
+  let foundKey = false;
 
   if (!(kid && keys)) {
-    return null;
+    throw new Error('keys or kid is missing');
```

Comment:
It would be helpful to have two different errors. Distinguish between 'kid' and 'keys'.